src/qthotkey/meson.build

```diff
@@ -1,17 +1,18 @@
 if get_option('qt6')
-  qtx11extras_dep = dependency('qt6', version: qt_req, required: false, modules: ['X11Extras'])
+  qtprivate_dep = dependency('qt6', version: qt_req, private_headers: true, required: false, modules: ['Core', 'Gui'])
 else
-  qtx11extras_dep = dependency('qt5', version: qt_req, required: false, modules: ['X11Extras'])
+  qtprivate_dep = dependency('qt5', version: qt_req, required: false, modules: ['X11Extras'])
 endif
 
-have_qthotkey = qtx11extras_dep.found()
+# Requires private headers and a UNIX system for X11 support
+have_qthotkey = qtprivate_dep.found() and target_machine.system() not in ['windows', 'darwin']
```

Comment:
You can reuse `have_darwin` and `have_windows` from the root meson.build file here.